R/projected.R

```diff
@@ -115,14 +115,16 @@ is.projectedCRS <- function(obj) {
 	if (is.null(wkt2) && is.na(p4str))
 		as.logical(NA)
 	else if (get("evolution_status", envir=.spOptions) == 2L) {
-		if (!requireNamespace("sf", quietly = TRUE))
-			stop("sf required for evolution_status==2L")
-		o <- try(sf::st_is_longlat(obj), silent=TRUE)
+		if (requireNamespace("sf", quietly = TRUE)) {
+		    o <- try(sf::st_is_longlat(obj), silent=TRUE)
 # rbgm workaround for +proj=utm +zone=18 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs +# +a=6378137.0 +es=0.006694380022900787 +lon_0=-75d00 +lat_0=0d00 +x_0=500000.0 +y_0=0.0 +k=0.9996 in bgmfiles Final_CAM_Boxes_8.bgm
-                if (inherits(o, "try-error"))
-                    length(grep("longlat", p4str, fixed = TRUE)) == 0L
-                else
-                    !o
+                    if (inherits(o, "try-error"))
+                        length(grep("longlat", p4str, fixed = TRUE)) == 0L
+                    else
+                        !o
+                } else {
+                    warning("Package sf not available")
```

Comment:
The non-sf case would end up returning a character string (the warning message) rather than a logical as `is.projected()` should; not sure if this was intended?
Shouldn't the grep-for-longlat-in-p4str fallback be used in the non-sf case as well, maybe returning NA if is.na(p4str) || !nzchar(p4str)?

Reply:
I don't think so; I need to check what it returns if anything at all, but warnings are not IIUC returned as strings, they are queued for display depending on the general option setting.

Reply:
I believe that today's two commits relieve the problem for the mentioned packages. I'll go ahead and re-run revdeps `"most"` for `sp` and the retiring packages - some packages took `sp` as given if retiring packages were strong dependencies. Will report back with rev-dep outcomes run under `_R_CHECK_SUGGESTS_ONLY_=true`.